include/swift/SIL/SILDebugScope.h

```diff
@@ -66,6 +66,28 @@ class SILDebugScope : public SILAllocated<SILDebugScope> {
   /// into.
   SILFunction *getParentFunction() const;
 
+  /// If this is a debug scope associated with an inlined call site, return the
+  /// SILLocation associated with the call site resulting from the final
+  /// inlining.
+  ///
+  /// This allows one to emit diagnostics based off of inlined code's final
+  /// location in the function that was inlined into.
+  SILLocation getOutermostInlineLocation() const {
+    if (!InlinedCallSite)
+      return SILLocation::invalid();
+
+    auto *scope = this;
+    do {
+      scope = scope->InlinedCallSite;
+    } while (scope->InlinedCallSite);
+
+    SILLocation callSite = scope->Loc;
+    if (callSite.isNull() || !callSite.isASTNode())
```

Comment:
I think the `!callSite.isASTNode()` condition causes this to not handle textual .sil files correctly, where all source locations are textual, since there is no AST. Is this check really necessary?

Reply:
I see your point. I imagine I am going to hit this when I roll out my SIL tests for this. I am going to loop back around in a little bit to fix this.